### ISSUE TITLE
Locking screen with WSP active causes clutter_actor_dispose: assertion failed: (priv->parent == NULL)

### ISSUE THREAD

**Issue by sfllaw** [G-dH/windows-search-provider#6]
I have version 17 of WSP installed from extensions.gnome.org on GNOME 45.2.

When I lock the screen (or suspend which also locks the screen), gnome-shell crashes with:
```
Dec 19 19:13:37 sfllaw-framework gnome-shell[2613]: **
Dec 19 19:13:37 sfllaw-framework gnome-shell[2613]: Clutter:ERROR:../clutter/clutter/clutter-actor.c:5480:clutter_actor_dispose: assertion failed: (priv->parent == NULL)
Dec 19 19:13:37 sfllaw-framework gnome-shell[2613]: Bail out! Clutter:ERROR:../clutter/clutter/clutter-actor.c:5480:clutter_actor_dispose: assertion failed: (priv->parent == NULL)
Dec 19 19:13:52 sfllaw-framework gnome-shell[3135]: (EE) failed to write to Xwayland fd: Broken pipe
```

If I disable WSP, this stops happening.

**Comment by sfllaw** [comment]
This sure resembles https://bugs.launchpad.net/ubuntu/+source/gnome-shell-extension-ubuntu-dock/+bug/2056506 where using `add_child` instead of `set_child` is what caused the problem.

See https://github.com/micheleg/dash-to-dock/commit/255ef10db9c330643188faca5ea82093f7ad0bb0 for the patch that fixed dash-to-dock. It seems that this is a problem with GNOME 45.

**Comment by G-dH** [comment]
Thanks for reporting this. Does disabling the `WSP Icon Position` option help?

**Comment by rAv3Nb0unD** [comment]
Same problem on Gentoo Linux, 6.6.58 kernel (LTS), Gnome 45.5. It crashes gnome shell and i have to log in back into the session. (I am on Wayland)

**Comment by rAv3Nb0unD** [comment]
I can run some tests now, What are the steps to reproduce in order to avoid this issue?

**Comment by rAv3Nb0unD** [comment]
One more behavior. It crashes when I disable this extension In the "Extensions"

**Comment by G-dH** [comment]
WSP Settings window > `WSP Icon Position` > `Hide`

**Comment by rAv3Nb0unD** [comment]
I set the "WSP Icon position" option to Hide. It crashed again, but on the next login screen locking works without issue

**Comment by rAv3Nb0unD** [comment]
Can you update this extension with this parameter off by default after install, or fix the behavior of this param?

**Comment by G-dH** [comment]
Thanks for the feedback.

**Comment by G-dH** [comment]
Fixed in WSP version 47.1, already available on the e.g.o website.
